public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -786,51 +785,86 @@ public SurveyQuestionResult apply(AchillesResult o) {
         List<DomainInfo> domainInfos =  ImmutableList.copyOf(domainInfoDao.findDomainTotals(getTests, getOrders));
         List<SurveyModule> surveyModules = ImmutableList.copyOf(surveyModuleDao.findByCanShowNotOrderByOrderNumberAsc(0));
 
+        if (domainInfos == null || surveyModules == null) {
+            throw new DataNotFoundException("No domain metadata available");
+        }
+
         DomainInfosAndSurveyModulesResponse response = new DomainInfosAndSurveyModulesResponse();
         response.setDomainInfos(domainInfos.stream()
                 .map(DomainInfo.TO_CLIENT_DOMAIN_INFO)
                 .collect(Collectors.toList()));
         response.setSurveyModules(surveyModules.stream()
                 .map(SurveyModule.TO_CLIENT_SURVEY_MODULE)
                 .collect(Collectors.toList()));
-
         return ResponseEntity.ok(response);
     }
 
     @Override
     public ResponseEntity<org.pmiops.workbench.model.CdrVersion> getCdrVersionUsed() {
-        CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         CdrVersion cdrVersion = cdrVersionDao.findByIsDefault(true);
+        if (cdrVersion == null) {
+            throw new ServerErrorException("No data present in table cdr version");
```

Comment:
This should probably be the exception above. 
`new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");`

Reply:
Yes changed thanks.